tests/test_cast_to_iamc.py

```diff
@@ -4,43 +4,40 @@
 
 from conftest import TEST_DTS
 
-def test_cast_from_value_col(meta_df):
+
+def test_cast_from_value_col(test_df_year):
     df_with_value_cols = pd.DataFrame([
-        ['model_a', 'scen_a', 'World', 'EJ/y', TEST_DTS[0], 1, 0.5],
-        ['model_a', 'scen_a', 'World', 'EJ/y', TEST_DTS[1], 6., 3],
-        ['model_a', 'scen_b', 'World', 'EJ/y', TEST_DTS[0], 2, None],
-        ['model_a', 'scen_b', 'World', 'EJ/y', TEST_DTS[1], 7, None]
+        ['model_a', 'scen_a', 'World', 'EJ/y', 2005, 1, 0.5],
```

Comment:
Is there a reason for preferring hardcoded values to referenced variables?

Reply:
Same comments below as well

Reply:
the previous implementation tested the same thing twice

0. create an `IamDataFrame` with year-columns
1. test it against the test-data with year column
2. test it again against the datetime-data converted to year columns, which is exactly the same object as 1.

So I simplified it by removing step 2, and then choose the shortest notation:
 `datetime(2005, 6, 17)` > `TEST_DTS[0]` > `2005`